internal/terraform/exec/exec_mock.go

```diff
@@ -66,33 +72,126 @@ func (mc *MockQueue) NextMockItem() *MockItem {
 	return mi
 }
 
-func MockExecutor(md MockItemDispenser) ExecutorFactory {
-	return func(_ string) *Executor {
-		if md == nil {
-			md = &MockCall{
-				MockError: "no mocks provided",
+type mockExecutor struct {
+	path    string
+	timeout time.Duration
+	md      MockItemDispenser
+}
+
+func (*mockExecutor) SetWorkdir(string) {}
+
+func (*mockExecutor) SetExecLogPath(string) {}
+
+func (*mockExecutor) SetLogger(*log.Logger) {}
+
+func (m *mockExecutor) SetTimeout(timeout time.Duration) {
+	m.timeout = timeout
+}
+
+func (m *mockExecutor) GetExecPath() string {
+	return m.path
+}
+
+func (m *mockExecutor) Version(ctx context.Context) (string, error) {
+	out, err := m.run(ctx, "version")
+	if err != nil {
+		return "", fmt.Errorf("failed to get version: %w", err)
+	}
+	outString := string(out)
+	lines := strings.Split(outString, "\n")
+	if len(lines) < 1 {
+		return "", fmt.Errorf("unexpected version output: %q", outString)
+	}
+	version := strings.TrimPrefix(lines[0], "Terraform v")
+
+	return version, nil
+}
+
+func (m *mockExecutor) Format(ctx context.Context, input []byte) ([]byte, error) {
+	return m.run(ctx, "fmt", "-")
+}
+
+func (m *mockExecutor) ProviderSchemas(ctx context.Context) (*tfjson.ProviderSchemas, error) {
+	outBytes, err := m.run(ctx, "providers", "schema", "-json")
+	if err != nil {
+		return nil, fmt.Errorf("failed to get schemas: %w", err)
+	}
+
+	var schemas tfjson.ProviderSchemas
+	err = json.Unmarshal(outBytes, &schemas)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal response: %w", err)
+	}
+
+	return &schemas, nil
+}
+
+func (m *mockExecutor) FormatterForVersion(v string) (Formatter, error) {
+	return formatterForVersion(v, m.Format)
+}
+
+func (m *mockExecutor) run(ctx context.Context, args ...string) ([]byte, error) {
```

Comment:
the `mockExecutor` is essentially a stripped down copy of the previous "melded" implementation we had before